modules/gapi/test/util/variant_tests.cpp

```diff
@@ -486,4 +500,209 @@ TEST(Variant, EXT_IndexOf)
+
+    var = char{'a'};
+    EXPECT_TRUE(cv::util::visit(visitor, var));
+    EXPECT_EQ(ss.str(), std::string("0:42,1:1,2:a,"));
+
+    var = float{6.0};
+    EXPECT_TRUE(cv::util::visit(visitor, var));
+    EXPECT_EQ(ss.str(), std::string("0:42,1:1,2:a,3:6,"));
+
+    var = test_validation::MyType{};
+    EXPECT_TRUE(cv::util::visit(visitor, var));
+    EXPECT_EQ(ss.str(), std::string("0:42,1:1,2:a,3:6,4:MyType,"));
+
+    var = test_validation::MyClass{};
+    EXPECT_TRUE(cv::util::visit(visitor, var));
+    EXPECT_EQ(ss.str(), std::string("0:42,1:1,2:a,3:6,4:MyType,5:MyClass,"));
+}
+
+TEST(Variant, StaticVisitor)
+{
+    using V = cv::util::variant<int, double, char, float, test_validation::MyType, test_validation::MyClass>;
+    V var{42};
+    std::stringstream ss;
+    test_validation::MyNoParamStaticVisitor visitor(ss);
+
+    cv::util::visit(visitor, var);
+    EXPECT_EQ(ss.str(), std::string("42,"));
+
+    var = double{1.0};
+    cv::util::visit(visitor, var);
+    EXPECT_EQ(ss.str(), std::string("42,1,"));
+
+    var = char{'a'};
+    cv::util::visit(visitor, var);
+    EXPECT_EQ(ss.str(), std::string("42,1,a,"));
+
+    var = float{6.0};
+    cv::util::visit(visitor, var);
+    EXPECT_EQ(ss.str(), std::string("42,1,a,6,"));
+
+    var = test_validation::MyType{};
+    cv::util::visit(visitor, var);
+    EXPECT_EQ(ss.str(), std::string("42,1,a,6,MyType,"));
+
+    var = test_validation::MyClass{};
+    cv::util::visit(visitor, var);
+    EXPECT_EQ(ss.str(), std::string("42,1,a,6,MyType,MyClass,"));
+}
+
+TEST(Variant, StaticIndexedVisitor)
+{
+    using V = cv::util::variant<int, double, char, float, test_validation::MyType, test_validation::MyClass>;
+    V var{42};
+
+    std::stringstream ss;
+    cv::util::visit(test_validation::MyNoParamStaticIndexedVisitor {ss}, var);
+    EXPECT_EQ(ss.str(), std::string("0:42,"));
+
+    var = double{1.0};
+    cv::util::visit(test_validation::MyNoParamStaticIndexedVisitor (ss), var);
+    EXPECT_EQ(ss.str(), std::string("0:42,1:1,"));
+
+    var = char{'a'};
+    cv::util::visit(test_validation::MyNoParamStaticIndexedVisitor (ss), var);
+    EXPECT_EQ(ss.str(), std::string("0:42,1:1,2:a,"));
+
+    var = float{6.0};
+    cv::util::visit(test_validation::MyNoParamStaticIndexedVisitor (ss), var);
+    EXPECT_EQ(ss.str(), std::string("0:42,1:1,2:a,3:6,"));
+
+    var = test_validation::MyType{};
+    cv::util::visit(test_validation::MyNoParamStaticIndexedVisitor (ss), var);
+    EXPECT_EQ(ss.str(), std::string("0:42,1:1,2:a,3:6,4:MyType,"));
+
+    var = test_validation::MyClass{};
+    cv::util::visit(test_validation::MyNoParamStaticIndexedVisitor (ss), var);
+    EXPECT_EQ(ss.str(), std::string("0:42,1:1,2:a,3:6,4:MyType,5:MyClass,"));
+}
+
+
+TEST(Variant, LambdaVisitor)
+{
+    using V = cv::util::variant<int, double, char, float, test_validation::MyType, test_validation::MyClass>;
+    V var{42};
+    {
+        cv::util::visit(cv::overload_lambdas(
```

Comment:
is this `cv::overload_lambdas` thing mandatory here? can `cv::util::visit` overload accept just a plain list of lambdas?

Reply:
..and construct this `cv::overload_lambdas` somehow internally..

Also, I doubt if this `overload_lambdas` should be in `cv::`, likely it should go to the `cv::util::`.

Reply:
it make sense
i'll try to add such overload, but not sure about Microsoft Windows Compiler because it failed for less complicated constructions before